code/controllers/subsystems/inventory.dm

```diff
@@ -16,7 +16,7 @@ SUBSYSTEM_DEF(inventory)
 		slots[S.id] = S
 
 /datum/controller/subsystem/inventory/proc/get_slot_datum(slot)
-	return slots.len >= slot ? slots[slot] : null
+	return (slots && slots.len >= slot) ? slots[slot] : null
```

Comment:
That proc shouldn't be called before `Initialize`. `if(slots)` check is redundant and harmful.

Reply:
It can be called before since it used to dress doll in pref window which can be opened while game is initializing